src/sentry/static/sentry/app/views/eventsV2/table/tableView.tsx

```diff
@@ -39,37 +35,8 @@ export type TableViewProps = {
  * is coupled to the `Location` object and derives its state entirely from it.
  * It implements methods to mutate the column state in `Location.query`.
  */
-class TableView extends React.Component<TableViewProps, TableState> {
-  constructor(props) {
-    super(props);
-
-    this.setState = () => {
-      throw new Error(
-        'TableView: Please do not directly mutate the state of TableView. Please read the comments on TableView.createColumn for more info.'
-      );
-    };
-  }
-
-  state = {
-    columnOrder: [],
-    columnSortBy: [],
-  } as TableState;
-
-  static getDerivedStateFromProps(props: TableViewProps): TableState {
-    // Avoid using props.location to get derived state.
-    const {eventView} = props;
-
-    return {
-      columnOrder: decodeColumnOrder({
-        field: eventView.getFieldNames(),
-        fieldnames: eventView.getFieldTitles(),
-      }),
-      columnSortBy: decodeColumnSortBy({
-        sort: eventView.getDefaultSort(),
-      }),
-    };
-  }
-
+class TableView extends React.Component<TableViewProps> {
+  // TODO: update this docs
```

Comment:
Docs :smile:

Reply:
Done in https://github.com/getsentry/sentry/pull/15045/commits/4df8671907c9f1ba6c47c01b9db62ff9f92aa4e3